test/extended/images/hardprune.go

```diff
@@ -0,0 +1,314 @@
+	})
+
+	g.It("should prune orphaned blobs", func() {
+		oc.SetOutputDir(exutil.TestContext.OutputDir)
+		outSink := g.GinkgoWriter
+		registryURL, err := registryutil.GetDockerRegistryURL(oc)
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		cleanUp := NewCleanUpContainer(oc)
+		defer cleanUp.Run()
+
+		dClient, err := testutil.NewDockerClient()
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		_, err = RunHardPrune(oc)
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		baseImg1, imageId, err := BuildAndPushImageOfSizeWithDocker(oc, dClient, "a", "latest", testImageSize, 2, outSink, true, false)
+		o.Expect(err).NotTo(o.HaveOccurred())
+		cleanUp.AddImage(baseImg1, imageId, "")
+		baseImg1Spec := fmt.Sprintf("%s/%s/a:latest", registryURL, oc.Namespace())
+
+		baseImg2, imageId, err := BuildAndPushImageOfSizeWithDocker(oc, dClient, "b", "latest", testImageSize, 2, outSink, true, false)
+		o.Expect(err).NotTo(o.HaveOccurred())
+		cleanUp.AddImage(baseImg2, imageId, "")
+		baseImg2Spec := fmt.Sprintf("%s/%s/b:latest", registryURL, oc.Namespace())
+
+		baseImg3, imageId, err := BuildAndPushImageOfSizeWithDocker(oc, dClient, "c", "latest", testImageSize, 2, outSink, true, false)
+		o.Expect(err).NotTo(o.HaveOccurred())
+		cleanUp.AddImage(baseImg3, imageId, "")
+		baseImg3Spec := fmt.Sprintf("%s/%s/c:latest", registryURL, oc.Namespace())
+
+		baseImg4, imageId, err := BuildAndPushImageOfSizeWithDocker(oc, dClient, "a", "latest", testImageSize, 2, outSink, true, false)
+		o.Expect(err).NotTo(o.HaveOccurred())
+		cleanUp.AddImage(baseImg4, imageId, "")
+		//baseImg4Spec := fmt.Sprintf("%s/%s/a:new", registryURL, oc.Namespace())
+
+		childImg1, imageId, err := BuildAndPushChildImage(oc, dClient, baseImg1Spec, "c", "latest", 1, outSink, true)
+		o.Expect(err).NotTo(o.HaveOccurred())
+		cleanUp.AddImage(childImg1, "", "")
+		childImg2, imageId, err := BuildAndPushChildImage(oc, dClient, baseImg2Spec, "b", "latest", 1, outSink, true)
+		o.Expect(err).NotTo(o.HaveOccurred())
+		cleanUp.AddImage(childImg2, "", "")
+		childImg3, imageId, err := BuildAndPushChildImage(oc, dClient, baseImg3Spec, "c", "latest", 1, outSink, true)
+		o.Expect(err).NotTo(o.HaveOccurred())
+		cleanUp.AddImage(childImg3, "", "")
+
+		err = oc.Run("tag").Args("--source=istag", "a:latest", "a-tagged:latest").Execute()
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		imgs := map[string]*imageapi.Image{}
+		for _, imgName := range []string{baseImg1, baseImg2, baseImg3, baseImg4, childImg1, childImg2, childImg3} {
+			img, err := oc.AsAdmin().Client().Images().Get(imgName, metav1.GetOptions{})
+			o.Expect(err).NotTo(o.HaveOccurred())
+			imgs[imgName] = img
+			o.Expect(img.DockerImageManifestMediaType).To(o.Equal(schema2.MediaTypeManifest))
+		}
+
+		deleted, err := RunHardPrune(oc)
+		o.Expect(err).NotTo(o.HaveOccurred())
+		// this shouldn't delete anything
+		o.Expect(deleted.Len()).To(o.Equal(0))
+
+		/* TODO: use a persistent storage for the registry to preserve data across re-deployments
```

Comment:
Sounds reasonable. You'll want to figure out how to get claimable PVs on the CI environment - what OpenShift Ansible options need to be set? Do we need to set up the filesystem in a special way? We've got logic to set up a VG for EmptyDir mounts already on XFS so we've got a place in [`origin-ci-tool`](https://github.com/openshift/origin-ci-tool/blob/master/oct/ansible/oct/roles/docker/tasks/configure_openshift_storage.yml) to add more set up in case we need to.  OpenShift Ansible inventory changes would land in [`aos-cd-jobs`](https://github.com/openshift/aos-cd-jobs/blob/master/sjb/inventory/group_vars/OSEv3/general.yml).